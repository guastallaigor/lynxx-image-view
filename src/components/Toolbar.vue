<template>
  <header role="banner" title="Banner" aria-label="Header" class="navbar">
    <div class="nav-container">
      <router-link to="/" aria-label="Logo Image Link" title="Favorite Views" class="brand link"
        ><img src="@/assets/img/logo.png" alt="Logo" class="logo"
      /></router-link>
      <div class="spacer"></div>
      <nav role="navigation" class="nav-menu" aria-label="Navigation bar" title="Navigation Bar">
        <router-link to="/" aria-label="Home Link" title="Home" class="link">Home</router-link
        ><router-link to="/favorite-views" aria-label="Favorite Views Button" title="Favorite Views" class="button link"
          >Favorite Views</router-link
        >
      </nav>
    </div>
  </header>
</template>

<script lang="ts">
import { Component, Vue } from "vue-property-decorator";

@Component
export default class Toolbar extends Vue {}
</script>

<style lang="scss" scoped>
@mixin hover-effect() {
  transition: opacity 0.2s ease-in-out;

  &:hover {
    opacity: 0.75;
  }
}

.navbar {
  position: fixed;
  left: 0;
  top: 0;
  right: 0;
  bottom: auto;
  z-index: 2;
  padding: 0 40px;
  border-bottom: 1px solid $light-gray;
  backdrop-filter: saturate(180%) blur(7px);

  @include sm-and-down-alternative() {
    padding: 0;
  }

  .nav-container {
    display: flex;
    max-width: 1200px;
    align-items: center;
    flex-flow: row wrap;
    margin: 0 auto;

    @include sm-and-down-alternative() {
      justify-content: center;
    }
  }

  .link {
    font-size: 13px;
    line-height: 16px;
    position: relative;
    vertical-align: top;
    text-decoration: none;
    padding: 20px;
    text-align: left;
    transition: opacity 300ms ease;
    display: inline-block;
    margin-right: 5px;
    margin-left: 5px;
    color: $dark-gray;
    background-color: transparent;
    font-weight: 400;
    @include hover-effect();
  }

  .brand {
    position: relative;
    line-height: 32px;

    @include sm-and-down-alternative() {
      line-height: 0.1em;
      padding: 0;
    }

    img {
      @include sm-and-down-alternative() {
        max-width: 200px;
      }
    }
  }

  .spacer {
    flex: 1;

    @include sm-and-down-alternative() {
      display: none;
    }
  }

  .nav-menu {
    display: flex;
    align-items: center;
    position: relative;
    justify-content: flex-end;
  }

  .button {
    margin-left: 10px;
    padding-top: 16px;
    padding-bottom: 16px;
    background-color: $yellow;
    border-radius: 4px;
    color: $dark-gray-2;
    font-weight: 400;
    @include hover-effect();
  }
}
</style>
