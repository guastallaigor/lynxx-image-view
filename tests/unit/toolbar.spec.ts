import { createLocalVue, shallowMount } from "@vue/test-utils";
import VueRouter from "vue-router";

import Toolbar from "@/components/Toolbar.vue";

describe("When I create the Toolbar component", () => {
  const createToolbarWrapper = () => {
    const localVue = createLocalVue();
    localVue.use(VueRouter);
    const router = new VueRouter();

    return shallowMount(Toolbar, {
      localVue,
      router
    });
  };

  it(`should exists`, () => {
    const wrapper = createToolbarWrapper();
    const toolbar = wrapper.find(".navbar");
    expect(wrapper.exists()).toBe(true);
    expect(toolbar.exists()).toBe(true);
    expect(toolbar.attributes().role).toBe("banner");
    expect(toolbar.attributes().title).toBe("Banner");
    expect(toolbar.attributes()["aria-label"]).toBe("Header");
  });

  it(`should have a image logo that it's a link`, () => {
    const wrapper = createToolbarWrapper();
    const link = wrapper.find(".nav-container>:nth-child(1)");
    expect(wrapper.exists()).toBe(true);
    expect(link.exists()).toBe(true);
    expect(link.attributes().to).toBe("/");
    expect(link.attributes().class).toContain("brand");
    expect(link.attributes().class).toContain("link");
    expect(link.attributes().title).toBe("Favorite Views");
    expect(link.attributes()["aria-label"]).toBe("Logo Image Link");
    const img = wrapper.find(".nav-container>:nth-child(1)>img");
    expect(img.exists()).toBe(true);
    expect(img.attributes().src).toContain("logo.png");
    expect(img.attributes().alt).toBe("Logo");
  });

  it(`should have a navbar that has a home link`, () => {
    const wrapper = createToolbarWrapper();
    expect(wrapper.exists()).toBe(true);
    const navMenu = wrapper.find(".nav-menu");
    expect(navMenu.exists()).toBe(true);
    expect(navMenu.attributes().role).toBe("navigation");
    expect(navMenu.attributes().title).toBe("Navigation Bar");
    expect(navMenu.attributes()["aria-label"]).toBe("Navigation bar");
  });

  it(`should have a favorite views button/link`, () => {
    const wrapper = createToolbarWrapper();
    expect(wrapper.exists()).toBe(true);
    const link = wrapper.find(".nav-menu>:nth-child(2)");
    expect(link.exists()).toBe(true);
    expect(link.attributes().to).toBe("/favorite-views");
    expect(link.attributes().class).toContain("link");
    expect(link.attributes().title).toBe("Favorite Views");
    expect(link.attributes()["aria-label"]).toBe("Favorite Views Button");
    expect(link.text()).toBe("Favorite Views");
  });

  it(`should have a home link`, () => {
    const wrapper = createToolbarWrapper();
    expect(wrapper.exists()).toBe(true);
    const link = wrapper.find(".nav-menu>:nth-child(1)");
    expect(link.exists()).toBe(true);
    expect(link.attributes().to).toBe("/");
    expect(link.attributes().class).toContain("link");
    expect(link.attributes().title).toBe("Home");
    expect(link.attributes()["aria-label"]).toBe("Home Link");
    expect(link.text()).toBe("Home");
  });

  it("should match snapshot", () => {
    const wrapper = createToolbarWrapper();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
