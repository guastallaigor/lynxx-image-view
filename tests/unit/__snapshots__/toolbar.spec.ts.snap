// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When I create the Toolbar component should match snapshot 1`] = `
<header role="banner" title="Banner" aria-label="Header" class="navbar">
  <div class="nav-container">
    <router-link-stub to="/" tag="a" ariacurrentvalue="page" event="click" aria-label="Logo Image Link" title="Favorite Views" class="brand link"><img src="@/assets/img/logo.png" alt="Logo" class="logo"></router-link-stub>
    <div class="spacer"></div>
    <nav role="navigation" aria-label="Navigation bar" title="Navigation Bar" class="nav-menu">
      <router-link-stub to="/" tag="a" ariacurrentvalue="page" event="click" aria-label="Home Link" title="Home" class="link">Home</router-link-stub>
      <router-link-stub to="/favorite-views" tag="a" ariacurrentvalue="page" event="click" aria-label="Favorite Views Button" title="Favorite Views" class="button link">Favorite Views</router-link-stub>
    </nav>
  </div>
</header>
`;
